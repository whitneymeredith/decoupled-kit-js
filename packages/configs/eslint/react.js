module.exports = {
	plugins: ['react'],
	env: {
		browser: true,
	},
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
