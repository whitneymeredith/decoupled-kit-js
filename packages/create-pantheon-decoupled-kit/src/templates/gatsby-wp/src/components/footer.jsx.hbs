import React from 'react'
import { Link, useStaticQuery, graphql } from 'gatsby'
{{#unless tailwindcss}}
import * as styles from './footer.module.css'
{{/unless}}

const Footer = () => {
	const menuQuery = useStaticQuery(graphql`
		query MenuQuery {
			wpMenu(name: { eq: "Example Menu" }) {
				id
				menuItems {
					nodes {
						id
						path
						label
					}
				}
			}
		}
	`)
	const nodes = menuQuery?.wpMenu?.menuItems?.nodes
		? menuQuery.wpMenu.menuItems.nodes
		: []

	const FooterMenu = () => (
		{{#if tailwindcss}}
		<nav className="flex flex-col max-w-lg mx-auto lg:max-w-screen-lg">
		{{else}}
		<nav className={styles.footerNav}>
		{{/if}}
			<ul>
				{nodes.map(({ id, label, path }, i) => {
					return (
						{{#if tailwindcss}}
						<li
							key={id + i}
							className="ml-3 list-disc text-blue-300 hover:text-blue-100"
						>
						{{else}}
						<li key={id + i} className={styles.navItem}>
						{{/if}}
						{{#if tailwindcss}}
							<Link
								className="text-blue-300 hover:text-blue-100 focus:text-purple-600 active:text-purple-300 "
								to={`/posts${path}`}
							>
								{label}
							</Link>
						{{else}}
							<Link to={`/posts${path}`}>{label}</Link>
						{{/if}}
						</li>
					)
				})}
			</ul>
		</nav>
	)

	return (
		{{#if tailwindcss}}
		<footer className="w-full text-white bg-black p-4 mt-12">
		{{else}}
		<footer className={styles.footer}>
		{{/if}}
			<FooterMenu />
			{{#if tailwindcss}}
			<div className="flex my-4 p-2">
				<span className="mx-auto">
					© {new Date().getFullYear()} Built with{' '}
					<a
						className="text-purple-300 underline hover:text-blue-100"
						href="https://www.gatsbyjs.com"
					>
						Gatsby
					</a>{' '}
					and{' '}
					<a
						className="text-blue-200 underline hover:text-blue-100"
						href="https://wordpress.org/"
					>
						WordPress
					</a>
				</span>
			</div>
			{{else}}
			<div className={styles.footerCopy}>
				<span>
					© {new Date().getFullYear()} Built with{' '}
					<a href="https://www.gatsbyjs.com">Gatsby</a> and{' '}
					<a href="https://wordpress.org/">WordPress</a>
				</span>
			</div>
			{{/if}}
		</footer>
	)
}

export default Footer
