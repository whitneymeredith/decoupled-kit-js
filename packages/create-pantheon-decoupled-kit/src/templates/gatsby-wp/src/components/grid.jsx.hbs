import React from 'react'
import { Link } from 'gatsby'
import { GatsbyImage } from 'gatsby-plugin-image'
{{#unless tailwindcss}}
import * as styles from './grid.module.css'
{{/unless}}

const GradientPlaceholder = () => {{#if tailwindcss}}(
	<div className="w-full h-full bg-gradient-to-b from-blue-100 to-blue-500" />
)
{{else}}
<div className={styles.gradientPlaceholder} />

{{/if}}
const withGrid = Component => {
	const GridedComponent = ({ data, ...props }) => {
		if (!data || !data.length) {
			return props.contentType ? (
				{{#if tailwindcss}}
				<h2 className="text-xl text-center mt-14">
					No {props.contentType} found üèú
				</h2>
				{{else}}
				<h2 className={styles.notFound}>No {props.contentType} found üèú</h2>
				{{/if}}
			) : null
		}

		return (
				{{#if tailwindcss}}
			<div
				className="mt-12 grid gap-5 max-w-content mx-auto lg:grid-cols-3 lg:max-w-screen-lg"
			>
				{{else}}
			<div className={styles.container}>
				{{/if}}
				{data.map((content, i) => {
					return <Component key={i} content={content} {...props} />
				})}
			</div>
		)
	}

	return GridedComponent
}

const GridItem = ({ href, imgSrc, altText, title }) => {
	return (
		{{#if tailwindcss}}
		<Link to={href} className="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500">
				<div className="flex-shrink-0 relative h-40">
			{{else}}
		<Link to={href} className={styles.card}>
			<div className={styles.cardImg}>
			{{/if}}
				{imgSrc ? (
					<GatsbyImage
						image={imgSrc}
						{{#if tailwindcss}}
						className="h-full w-full"
						{{else}}
						style=\{{ height: '100%', width: '100%' }}
						{{/if}}
						objectFit="fit"
						alt={altText}
					/>
				) : (
					<GradientPlaceholder />
				)}
			</div>
			{{#if tailwindcss}} 
			<h2 className="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900">
			{{else}}
			<h2 className={styles.cardTitle}>{{/if}}{title} &rarr;</h2>
		</Link>
	)
}

const PostGridItem = ({ content: { post } }) => {
	const imgSrc =
		post?.featuredImage?.node.localFile.childImageSharp.gatsbyImageData || null
	const altText = post?.featuredImage?.node.altText || post.title

	return (
		<GridItem
			href={`/posts${post.uri}`}
			imgSrc={imgSrc}
			altText={altText}
			title={post.title}
		/>
	)
}

const PageGridItem = ({ content: { page } }) => {
	const imgSrc =
		page?.featuredImage?.node.localFile.childImageSharp.gatsbyImageData || null
	const altText = page?.featuredImage?.node.altText || page.title

	return (
		<GridItem
			href={`/pages${page.uri}`}
			imgSrc={imgSrc}
			altText={altText}
			title={page.title}
		/>
	)
}

export const PostGrid = withGrid(PostGridItem)
export const PageGrid = withGrid(PageGridItem)
