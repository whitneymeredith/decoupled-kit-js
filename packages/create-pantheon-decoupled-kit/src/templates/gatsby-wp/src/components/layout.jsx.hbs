import React from 'react'
import { Link } from 'gatsby'
import Footer from './footer'
{{#unless tailwindcss}}
import * as styles from './layout.module.css'
{{/unless}}

const Layout = ({ isHomePage, children }) => {
	return (
		{{#if tailwindcss}}
		<div className="min-h-screen max-h-screen min-w-screen max-w-screen flex flex-col" data-is-root-path={isHomePage}>
			<nav className="my-0 pt-10 px-5 text-xl">
				<ul className="flex flex-row flex-wrap sm:flex-nowrap list-none justify-between max-w-screen-sm mx-auto">
		{{else}}
		<div className={styles.layout} data-is-root-path={isHomePage}>
			<nav>
				<ul className={styles.navLinks}>
		{{/if}}
					{[
						['🏠 Home', '/'],
						['📰 Posts', '/posts'],
						['📑 Pages', '/pages'],
						['⚛️ Examples', '/examples'],
					].map(([title, href], i) => (
						<li{{#if tailwindcss}} className={`${href === '/' ? 'mr-auto' : 'mx-4'}`}{{/if}} key={`key-${i}`}>
							<Link{{#if tailwindcss}} className="font-sans hover:underline"{{/if}} to={href}>{title}</Link>
						</li>
					))}
				</ul>
			</nav>
			<main{{#if tailwindcss}} className="mb-auto"{{/if}}>{children}</main>

			<Footer />
		</div>
	)
}

export default Layout
