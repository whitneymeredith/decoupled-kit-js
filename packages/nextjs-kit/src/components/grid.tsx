import React from 'react';

/**
 *
 * @param props.cols - The number of columns in the grid, this is an optional prop which will default to 3 if not provided
 * @param props.children - The JSX elements used as the content of the grid
 * @returns A style and positioning helper grid component that can be used with the withGrid HOC component
 */

export const Grid = ({
	cols = 3,
	children,
}: {
	cols?: number;
	children?: JSX.Element[];
}) => {
	return (
		<div
			className={`ps-mt-12 ps-grid ps-gap-5 ps-max-w-content ps-mx-auto lg:ps-max-w-screen-lg lg:ps-grid-cols-${String(
				cols,
			)}`}
		>
			{children}
		</div>
	);
};

/**
 *
 * @param Component - A component that takes in content that is to be displayed on the grid
 * @remarks
 * The Component used must accept the data to be displayed as `content` to function properly
 * @example
 * ```
 * const ArticleCard= ({ content }) => {
 * 	return (
 * 		<div>
 * 			<h2>{content.title}</h2>
 * 			<div>{content.excerpt}</div>
 * 		</div>
 * 	)
 * }
 * ```
 * @returns A Higher Order Component that returns the data mapped to the Component in a grid
 * @example
 * ```
 * const MyPage = ({ myArticles }) => {
 *  const ArticleGrid = withGrid(ArticleCard)
 * 	 return (
 * 		 <>
 * 			 <ArticleGrid
 * 				 data={myArticles}
 * 				 cols={4}
 * 				 FallbackComponent={<span>No Data Found</span>}
 * 			 />
 * 		 </>
 * 	 )
 * }
 * ```
 */
export const withGrid = (Component: React.ElementType) => {
	/**
	 * @param props.data - The to be passed to the Component as the content prop
	 * @param props.cols - The number of columns to build a grid with. This is an optional prop which will default to 3 if not provided
	 * @default 3
	 * @param props.FallbackComponent -  Component to be rendered if data is undefined
	 * @returns The component passed to withGrid in a grid with the given number of columns
	 */
	const GridedComponents = <Type extends object>({
		data,
		cols,
		FallbackComponent,
		...props
	}: {
		data?: Type[];
		cols?: number;
		FallbackComponent?: React.ElementType;
	}): JSX.Element => {
		return (
			<>
				{data ? (
					<Grid cols={cols}>
						{data.map((content, i) => {
							return <Component key={i} content={content} {...props} />;
						})}
					</Grid>
				) : FallbackComponent ? (
					<FallbackComponent />
				) : null}
			</>
		);
	};

	return GridedComponents;
};
