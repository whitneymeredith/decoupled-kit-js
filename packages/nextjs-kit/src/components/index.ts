export * from './header';
export * from './footer';
export * from './paginator';
export * from './contentWithImage';
export * from './recipe';
export * from './grid';
export * from './previewRibbon';
