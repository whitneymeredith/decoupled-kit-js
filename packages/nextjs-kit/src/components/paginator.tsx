import React, { useState, useEffect } from 'react';
import { useRouter } from 'next/router';

/**
 * Options type for {@link Paginator}
 * @typeParam Type - type to use for the data passed in to be paginated
 */
export interface PaginationProps<Type> {
	/**
	 * The type of data to paginate
	 */
	data: Type[];
	/**
	 * Number of items per page
	 */
	itemsPerPage: number;
	/**
	 * * start: where to start the breakpoint
	 *
	 * end: where to end the breakpoint
	 *
	 * add: how many buttons to add when the breakpoint is clicked
	 *
	 * (`add` * x) + `start` = `end` where x is a number of clicks it takes to fill in all of the buttons
	 * For example: If there are 25 buttons and the start = 5 and end = 25, then add should be 5 or 10.
	 */
	breakpoints: {
		start: number;
		end: number;
		add: number;
	};
	/**
	 * If true, uses Next.js shallow routing {@link https://nextjs.org/docs/routing/shallow-routing}
	 */
	routing: boolean;
	/**
	 * The React component to render for each datum
	 */
	Component: React.ElementType;
}

/**
 * Type predicate to determine if an item is a number
 * @param {number | null | undefined} item an item
 * @returns true if the item is a number
 */
const isNumber = (item: number | null | undefined): item is number =>
	typeof item === 'number';

/**
 * Type predicate to determine if an item is a number
 * @param {unknown} element some type of HTMLElement
 * @returns true if the element is an HTMLElement
 */
const isHTMLElement = (element: unknown): element is HTMLElement => {
	return element instanceof HTMLElement;
};

/**
 * @param {PaginationProps} props - The props needed for the paginator component
 * @param props.data - An array of paginator objects
 * @param props.itemsPerPage - How many items to display per page
 * @param props.breakpoints - Breakpoints has 3 properties: start, end, and add. Set to an empty object for no breakpoint.
 * @param props.routing If true, shallow routing will be enabled. Check the examples/pagination route to see it in action
 * @param props.Component React Component that takes in currentItems as props and maps over them.
 * currentItems is a subset of `props.data`, so any component that works for data will work here.
 *
 * @see {@link https://github.com/pantheon-systems/decoupled-kit-js/tree/canary/starters/next-drupal-starter/pages/examples/pagination/[[...page]].js} for a full example implementation
 * @example
 * ```
 * <Paginator
 *   data={data}
 *   itemsPerPage={itemsPerPage}
 *   breakpoints={{ start: 6, end: 12, add: 6 }}
 *   routing
 *   Component={MyComponent}
 * />
 * ```
 *
 * @returns Component with data rendered by the passed in Component and page buttons
 */
export const Paginator = <Type extends object>({
	data,
	itemsPerPage,
	breakpoints,
	routing,
	Component,
}: PaginationProps<Type>) => {
	// configurable breakpoints
	// This value will be the start of the separator.
	const [breakStart, setBreakStart] = useState<number | null>(
		breakpoints?.start || null,
	);

	// This value will be the button to start with after separator
	const breakEnd = breakpoints?.end || null;
	// how many buttons to add when the separator is clicked
	const breakAdd = breakpoints?.add || null;

	const router = useRouter();
	// get current path from router.pathname
	// and trim off catchalls
	const currentRoute = router.pathname.replace(/\/{1}\[{1,2}.*\]{1,2}$/, '');
	const routeKey = Object.keys(router.query)[0];
	const [currentPageQuery, setCurrentPageQuery] = useState<number>(
		Number(router.query[routeKey]) || 1,
	);

	const [offset, setOffset] = useState<number>(
		(currentPageQuery - 1) * itemsPerPage,
	);
	const [currentItems, setCurrentItems] = useState<Type[]>([]);
	const [totalItems] = useState<number>(data.length);

	const [totalPages] = useState<number>(Math.ceil(data.length / itemsPerPage));

	// since we fetch ALL items, filtering the current items client side
	// further reduces API calls to the server
	useEffect(() => {
		setCurrentItems(
			data.slice(
				(currentPageQuery - 1) * itemsPerPage,
				itemsPerPage * currentPageQuery,
			),
		);
		// set the new offset
		const newOffset = ((currentPageQuery - 1) * itemsPerPage) % totalItems;
		if (currentPageQuery > 0) {
			setOffset(Number(newOffset));
		}

		routing &&
			router.push(
				`${currentRoute}/${currentPageQuery}`,
				`${currentRoute}/${currentPageQuery}`,
				{
					shallow: true,
				},
			);
	}, [data, offset, itemsPerPage, breakStart, currentPageQuery, totalItems]);

	// track window width to appropriately hide and show buttons on small viewports
	const [windowWidth, setWindowWidth] = useState<number>();
	useEffect(() => {
		const eventListener = () => {
			setWindowWidth(window.innerWidth);
		};
		setWindowWidth(window.innerWidth);
		window.addEventListener('resize', () => eventListener);

		return () => {
			window.removeEventListener('resize', eventListener);
		};
	}, []);

	const handlePageClick: React.MouseEventHandler<HTMLButtonElement> = (
		event,
	) => {
		if (!isHTMLElement(event.target)) return;
		if (event.target.id === 'next-btn') {
			// set new offset
			offset < totalItems - itemsPerPage &&
				setOffset(Number(offset) + Number(itemsPerPage));
			// set the current page
			currentPageQuery < totalPages &&
				setCurrentPageQuery(Number(currentPageQuery + 1));
		} else if (event.target.id === 'back-btn') {
			// set new offset
			offset >= 0 && setOffset(offset - itemsPerPage);
			// set the current page
			currentPageQuery > 1 && setCurrentPageQuery(Number(currentPageQuery - 1));
		} else {
			// the number of the page button clicked
			const clickedPage = Number(event.target.innerHTML);
			setCurrentPageQuery(clickedPage);
		}
	};

	const RenderButtons = () => {
		if (totalPages <= 1) {
			// if there is only one page
			// don't render any buttons
			return null;
		}
		const buttons = [];

		// Create buttons given the number of
		// total pages
		for (let i = 0; i < totalPages; i++) {
			const pageNumber = Number(i + 1);
			const defaultButton = (
				<button
					className={`
          ${
						currentPageQuery === pageNumber
							? 'ps-block'
							: 'ps-hidden md:ps-block'
					}
          ps-h-16 ps-w-12 ps-border-t-2 ps-border-b-2 ps-border-black ps-bg-white hover:ps-bg-blue-300 focus:ps-bg-blue-200 focus:ps-border-blue-300 ${
						currentPageQuery === pageNumber
							? 'ps-border-blue-700 ps-border-2'
							: ''
					}
          `}
					onClick={handlePageClick}
					key={pageNumber}
					aria-label={
						currentPageQuery === pageNumber
							? `Current Page, Page ${pageNumber}`
							: `Go to Page ${pageNumber}`
					}
					aria-current={currentPageQuery === pageNumber ? true : false}
				>
					{pageNumber}
				</button>
			);

			// separator button
			if (i === breakStart) {
				if (isNumber(breakAdd)) {
					if (breakStart + breakAdd >= totalPages) {
						buttons.push(defaultButton);
						continue;
					}
				}
				buttons.push(
					<button
						className={`ps-hidden md:ps-block ps-h-16 ps-w-12 ps-border-2 ps-border-black ps-bg-slate-200 hover:ps-bg-blue-300 focus:ps-bg-blue-200 focus:ps-border-blue-300`}
						onClick={() => {
							if (isNumber(breakAdd)) {
								setBreakStart(breakStart + breakAdd);
							}
						}}
						key={'...'}
						aria-label="Expand Hidden Buttons"
					>
						...
					</button>,
				);
			}
			// if we have a breakStart, don't render the middle buttons
			if (isNumber(breakStart) && isNumber(breakEnd)) {
				if (pageNumber >= breakStart && pageNumber < breakEnd) {
					if (isNumber(windowWidth)) {
						if (windowWidth < 768) {
							buttons.push(defaultButton);
						}
						continue;
					}
				}
			}
			buttons.push(defaultButton);
		}
		const sharedNextAndBackBtnStyles =
			'ps-h-16 ps-w-12 disabled:ps-bg-gray-500 hover:ps-bg-blue-300 focus:ps-bg-blue-200 focus:ps-border-blue-300 ps-border-y-2 ps-border-black ps-bg-white';
		// returns the row of buttons
		return (
			<nav role="navigation" aria-label="Pagination Navigation">
				{/* back button */}
				<ul className="ps-list-none ps-flex ps-flex-row ps-justify-center ps-mx-auto ps-mt-auto ps-mb-4 [&>li]:ps-p-0">
					<li key="back">
						{/* TODO: For improved accessibility, these buttons should be refactored to anchor tags. */}
						<button
							className={`${sharedNextAndBackBtnStyles} ps-border-l-2`}
							id="back-btn"
							disabled={offset === 0}
							onClick={handlePageClick}
							aria-label="Go to Previous Page"
						>
							{'<'}
						</button>
					</li>
					{/* map buttons[] */}
					{buttons.map((btn, i) => (
						<li key={i}>{btn}</li>
					))}
					{/* next button */}
					<li key="next">
						<button
							className={`${sharedNextAndBackBtnStyles} ps-border-r-2`}
							id="next-btn"
							disabled={offset >= totalItems - itemsPerPage}
							onClick={handlePageClick}
							aria-label="Go to Next Page"
						>
							{'>'}
						</button>
					</li>
				</ul>
			</nav>
		);
	};
	return (
		<div className="ps-max-w-full">
			{totalPages > 1 ? (
				<h3 className="ps-mb-8 ps-prose-sm ps-font-bold">
					Page {currentPageQuery}/{totalPages}
				</h3>
			) : null}
			<section>
				{/* Component passed in that will render the data */}
				<Component currentItems={currentItems} />
			</section>
			<div className="ps-sticky lg:ps-bottom-12 ps-bottom-4 ps-mt-10">
				<RenderButtons />
			</div>
		</div>
	);
};
