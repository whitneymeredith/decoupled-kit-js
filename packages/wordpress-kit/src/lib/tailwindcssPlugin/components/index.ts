import { ImageComponent } from './Image';
import { PullQuoteComponent } from './PullQuote';
import { TableComponent } from './Table';
import { Quote } from './Quote';
import { GalleryComponent } from './Gallery';
import { AudioComponent } from './Audio';
import { CoverComponent } from './Cover';
import { mediaAndText } from './MediaAndText';
import { ButtonsComponent } from './Buttons';
import { FileMediaComponent } from './FileMedia';
import { VideoComponent } from './Video';
import { SpacerComponent } from './Spacer';
import { SeparatorComponent } from './Separator';
import { ColumnsComponent } from './Columns';

export {
	ImageComponent,
	PullQuoteComponent,
	TableComponent,
	Quote,
	GalleryComponent,
	AudioComponent,
	CoverComponent,
	mediaAndText,
	ButtonsComponent,
	FileMediaComponent,
	VideoComponent,
	SpacerComponent,
	SeparatorComponent,
	ColumnsComponent,
};
