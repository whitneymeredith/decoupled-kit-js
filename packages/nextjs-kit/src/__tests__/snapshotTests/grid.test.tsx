import { render } from '@testing-library/react';

import { withGrid } from '../../components/grid';
import exampleGridData from '../data/exampleGridData.json';

/**
 * @vitest-environment jsdom
 */

interface GridifyProps {
	content: {
		title: string;
		body: string;
	};
}

const Gridify: React.FC<GridifyProps> = ({
	content,
}: GridifyProps): JSX.Element => {
	return (
		<div>
			<h1>{content.title}</h1>
			<p>{content.body}</p>
		</div>
	);
};
const GridifyGrid = withGrid(Gridify);

const Fallback: React.FC = (): JSX.Element => {
	return (
		<>
			<h1>No Data</h1>
		</>
	);
};

describe('<Grid />', () => {
	it("should render 'Grid'", () => {
		const { asFragment } = render(<GridifyGrid data={exampleGridData} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it("should render a 'Grid' with 4 columns", () => {
		const { asFragment } = render(
			<GridifyGrid data={exampleGridData} cols={4} />,
		);

		expect(asFragment()).toMatchSnapshot();
	});
	it("should render FallbackComponent instead of 'Grid'", () => {
		const { asFragment } = render(<GridifyGrid FallbackComponent={Fallback} />);

		expect(asFragment()).toMatchSnapshot();
	});
});
