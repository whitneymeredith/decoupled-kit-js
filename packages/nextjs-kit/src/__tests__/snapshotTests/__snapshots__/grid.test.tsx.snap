// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Grid /> > should render 'Grid' 1`] = `
<DocumentFragment>
  <div
    class="ps-mt-12 ps-grid ps-gap-5 ps-max-w-content ps-mx-auto lg:ps-max-w-screen-lg lg:ps-grid-cols-3"
  >
    <div>
      <h1>
        Item 1
      </h1>
      <p>
        First item in the grid
      </p>
    </div>
    <div>
      <h1>
        Item 2
      </h1>
      <p>
        Second item in the grid
      </p>
    </div>
    <div>
      <h1>
        Item 3
      </h1>
      <p>
        Third item in the grid
      </p>
    </div>
    <div>
      <h1>
        Item 4
      </h1>
      <p>
        Fourth item in the grid
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Grid /> > should render FallbackComponent instead of 'Grid' 1`] = `
<DocumentFragment>
  <h1>
    No Data
  </h1>
</DocumentFragment>
`;

exports[`<Grid /> > should render a 'Grid' with 4 columns 1`] = `
<DocumentFragment>
  <div
    class="ps-mt-12 ps-grid ps-gap-5 ps-max-w-content ps-mx-auto lg:ps-max-w-screen-lg lg:ps-grid-cols-4"
  >
    <div>
      <h1>
        Item 1
      </h1>
      <p>
        First item in the grid
      </p>
    </div>
    <div>
      <h1>
        Item 2
      </h1>
      <p>
        Second item in the grid
      </p>
    </div>
    <div>
      <h1>
        Item 3
      </h1>
      <p>
        Third item in the grid
      </p>
    </div>
    <div>
      <h1>
        Item 4
      </h1>
      <p>
        Fourth item in the grid
      </p>
    </div>
  </div>
</DocumentFragment>
`;
